### PR TITLE
Portfolio: preserve data in the browser

### PR DESCRIPTION
**Pull request by udev-21**
implemented all requirements according to assignment

### CODE REVIEW

**Review by FrederickMih — COMMENTED** [review]
Hi @udev-21 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

**This is a collaborative project hence both members of the team are supposed to actively work together and this is mainly observed via commit messages. Let the other team member actively collaborate.**

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by udev-21** [comment]
Hello @FrederickMih 
Even though we worked together on the above commits via zoom (but he didn't commit because we didn't have time to exchange our roles it took about 5-10 minutes).

**Comment by FrederickMih** [comment]
@udev-21, right now the only way I can monitor how each team member actively collaborates in a project is through commit messages.

File: index.js [hunk 1]
```diff
@@ -219,4 +219,33 @@ myForm.addEventListener("submit", (event) => {
...
+
+let formInputValues = {
+    firstname:"",
+    lastname:"",
+    email: "",
+    message: "",
+};
+
+if (!localStorage.getItem('form')) {
+    localStorage.setItem('form', JSON.stringify(formInputValues));
```
Comment (mateo951) [inline index.js:234]: The form data for all inputs should be stored in a single object; saved to a single local storage register 🙌

File: index.js [hunk 2]
```diff
@@ -219,4 +219,33 @@ myForm.addEventListener("submit", (event) => {
...
+}else {
+    formInputValues = JSON.parse(localStorage.getItem('form'))
```
Comment (mateo951) [inline index.js:236]: The student should use the `localStorage.getItem` and `localStorage.setItem` functions to save and retrieve the form data object to the local storage✔

Comment (mateo951) [inline index.js:236]: In order to be used after retrived from the local storage the saved string is turned back into an object using `JSON.parse` or similar methods 📝 Well done!

**Review by mateo951 — APPROVED** [review]
### Hi @udev-21, 

Be sure to get in touch with student success if you are having difficulties with your partners 🤓 Awesome work! Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**